internal/pkg/describe/status.go

```diff
@@ -341,6 +336,391 @@ func (a *apprunnerServiceStatus) HumanString() string {
+		shortID = shortTaskID(ts.ID)
+	}
+	statusString += fmt.Sprint(shortID)
+	statusString += fmt.Sprintf("\t%s", ts.LastStatus)
+
+	revision := "-"
+	v, err := awsecs.TaskDefinitionVersion(ts.TaskDefinition)
+	if err == nil {
+		revision = strconv.Itoa(v)
+	}
+	statusString += fmt.Sprintf("\t%s", revision)
+
+	startedSince := "-"
+	if !ts.StartedAt.IsZero() {
+		startedSince = humanizeTime(ts.StartedAt)
+	}
+	statusString += fmt.Sprintf("\t%s", startedSince)
+
+	if config.shouldShowCapProvider {
+		cp := "-"
+		if ts.CapacityProvider != "" {
+			cp = ts.CapacityProvider
+		}
+		statusString += fmt.Sprintf("\t%s", cp)
+	}
+
+	if config.shouldShowContainerHealth {
+		ch := "-"
+		if ts.Health != "" {
+			ch = ts.Health
+		}
+		statusString += fmt.Sprintf("\t%s", ch)
+	}
+	return statusString
+}
+
+type ecsTaskStatusConfigOpts func(config *ecsTaskStatusConfig)
+
+type ecsTaskStatusConfig struct {
+	shouldShowCapProvider     bool
+	shouldShowContainerHealth bool
+}
+
+func withCapProviderShown(config *ecsTaskStatusConfig) {
+	config.shouldShowCapProvider = true
+}
+
+func withContainerHealthShow(config *ecsTaskStatusConfig) {
+	config.shouldShowContainerHealth = true
+}
+
+type taskTargetHealth struct {
+	HealthStatus   elbv2.HealthStatus `json:"healthStatus"`
+	TaskID         string             `json:"taskID"`
+	TargetGroupARN string             `json:"targetGroup"`
+}
+
+// targetHealthForTasks finds the target health in a target group, if any, for each task.
+func targetHealthForTasks(targetsHealth []*elbv2.TargetHealth, tasks []*awsecs.Task, targetGroupARN string) []taskTargetHealth {
+	var out []taskTargetHealth
+
+	// Create a set of target health to be matched against the tasks' private IP addresses.
+	// An IP target's ID is the IP address.
+	targetsHealthSet := make(map[string]*elbv2.TargetHealth)
+	for _, th := range targetsHealth {
+		targetsHealthSet[th.TargetID()] = th
+	}
+
+	// For each task, check if it is a target by matching its private IP address against targetsHealthSet.
+	// If it is a target, we try to add it to the output.
+	for _, task := range tasks {
+		ip, err := task.PrivateIP()
+		if err != nil {
+			continue
+		}
+
+		// Check if the IP is a target
+		th, ok := targetsHealthSet[ip]
+		if !ok {
+			continue
+		}
+
+		if taskID, err := awsecs.TaskID(aws.StringValue(task.TaskArn)); err == nil {
+			out = append(out, taskTargetHealth{
+				TaskID:         taskID,
+				HealthStatus:   *th.HealthStatus(),
+				TargetGroupARN: targetGroupARN,
+			})
+		}
+	}
+
+	return out
+}
+
+type valueWithIndex struct {
+	value int
+	index int
+}
+
+// summaryBar returns a summary bar given data and the string representations of each data category.
+// For example, data[0] will be represented by representations[0] in the summary bar.
+// If len(representations) < len(data), the default representation "□" is used for all data category with missing representation.
+func summaryBar(data []int, representations []string) string {
```

Comment:
Just writing this down so that it's also in the back of your head :D but I think it'd be super cool if we can move this to components here: https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/term/progress/component.go in a later PR.

Maybe we need a `type SummaryBar struct` component that implements the `Render` interface.

It'd also allow us to unit test this component separately.